generators/client/templates/angular/src/main/webapp/app/core/auth/principal.service.ts.ejs

```diff
@@ -90,6 +96,13 @@ export class Principal {
                 <%_ if (websocket === 'spring-websocket') { _%>
                 this.trackerService.connect();
                 <%_ } _%>
+
+                <%_ if (enableTranslation && authenticationType !== 'oauth2') { _%>
+                // After retrieve the account info, the language will be changed to
+                // the user's preferred language configured in the account setting
+                this.languageService.changeLanguage(this.userIdentity.langKey);
```

Comment:
One problem I see is that account info might be fetched while not login in and you might already have changed language in UI, so we need to ensure this doesnt switch it back

Reply:
here, i don't get it. we only chang language when we fetche the account info. if user not login, it won't reach here, it will use the the native language.